server/src/main/java/org/elasticsearch/index/mapper/FieldTypeLookup.java

```diff
@@ -70,10 +69,12 @@
             }
         }
 
+        final Map<String, String> aliasToConcreteName = new HashMap<>();
         for (FieldAliasMapper fieldAliasMapper : fieldAliasMappers) {
             String aliasName = fieldAliasMapper.name();
             String path = fieldAliasMapper.path();
             aliasToConcreteName.put(aliasName, path);
+            fullNameToFieldType.put(aliasName, fullNameToFieldType.get(path));
         }
 
         this.dynamicKeyLookup = new DynamicKeyFieldTypeLookup(dynamicKeyMappers, aliasToConcreteName);
```

Comment:
I think we can extend this to DynamicKeyFieldTypeLookup as well?  Rather than passing the whole alias map, just check for each alias path to see if it matches an existing entry in the dynamicKeyMappers map, and if it does, add a new entry with the alias path and the original dynamic mapper.

Reply:
I was a bit nervous about adapting DynamicKeyFieldTypeLookup, I did not find a straight-forward replacement for the map, and I am happy to look as a followup